config/zephyr/Kconfig

```diff
@@ -34,6 +34,43 @@ menuconfig CHIP
 
 if CHIP
 
+config CHIP_DEVICE_VENDOR_ID
+	int "Device vendor ID"
+	default 9050 # 0x235A
+	range 0 65535
+	help
+	  Identifier of the device manufacturer, assigned by Connectivity Standards
+	  Alliance. It is used in various CHIP areas, such as the Basic Information
+	  cluster or OTA (Over-the-air update) image header.
+
+config CHIP_DEVICE_PRODUCT_ID
+	int "Device product ID"
+	default 0
+	range 0 65535
+	help
+	  Identifier of the product, assigned by the device manufacturer. It is used
+	  in various CHIP areas, such as the Basic Information cluster or OTA
+	  (Over-the-air update) image header.
+
+config CHIP_DEVICE_SOFTWARE_VERSION
+	int "Device software version"
+	default 0
+	range 0 4294967295
+	help
+	  A number identifying the software version. It is used in various CHIP
+	  areas, such as the Basic Information cluster or OTA (Over-the-air update)
+	  image header. Note that due to the rollback protection a device will only
+	  accept a software update whose version is greater than the current one.
+
+config CHIP_DEVICE_SOFTWARE_VERSION_STRING
+	string "Device software version string"
+	default "prerelease"
+	help
+	  A string between 1 and 64 characters that provides a user-friendly
+	  description of the numeric software version specified in
+	  CHIP_DEVICE_SOFTWARE_VERSION. It is used in various CHIP areas, such as
+	  the Basic Information cluster or OTA (Over-the-air update) image header.
```

Comment:
Question: Is this not redundant information which is also part of the ChipProjectConfig? If yes, is there a way to not introduce this redundancy. This leads to errors during releases.

Reply:
Yes, if you don't specify these values in `CHIPProjectConfig.h`, they will be automatically set based on the Kconfig values. Basically, having these values available at the CMake level allows us to set both the C-defines and call the Python script to generate the OTA image.

Reply:
Ok, thanks for the clarification 🙂